### PR TITLE
ARROW-12068: [Python] Stop using distutils

### PR DESCRIPTION
**Pull request by pitrou**
According to PEP 632, distutils will be deprecated in Python 3.10 and removed in 3.12.

* switch to `setuptools` for general packaging
* use the `Version` class from the `packaging` project instead of `distutils.LooseVersion`

### CODE REVIEW

**Comment by pitrou** [comment]
Triggered a crossbow build here: https://github.com/ursacomputing/crossbow/branches/all?query=build-119

**Comment by pitrou** [comment]
It looks like AppVeyor will require more debugging unfortunately :-(

**Comment by pitrou** [comment]
I triggered additional crossbow builds here: https://github.com/ursacomputing/crossbow/branches/all?query=build-120 and https://github.com/ursacomputing/crossbow/branches/all?query=build-121

File: python/pyarrow/vendored/version.py [hunk 1]
```diff
@@ -0,0 +1,545 @@
+# Vendored from https://github.com/pypa/packaging,
```
Comment (xhochy) [inline python/pyarrow/vendored/version.py:1]: You'll need to add this to the NOTICE file too

**Review by jorisvandenbossche — APPROVED** [review]
Thanks, looks good to me

**Pull request closed without merging**